libraries/AP_Camera/AP_Camera.cpp

```diff
@@ -242,9 +259,21 @@ bool AP_Camera::update_location(const struct Location &loc)
         // be called without a new GPS fix
         return false;
     }
+
     if (get_distance(loc, _last_location) < _trigg_dist) {
         return false;
     }
-    _last_location = loc;
-    return true;
+
+    if ((unsigned) ahrs.roll_sensor/100.0f > _max_roll) {
```

Comment:
The \100.0f is promoting it to float. Take the .0f off and use labs() for
integer.
On Nov 17, 2015 11:26 PM, "WickedShell" <notifications@github.com> wrote:

> In libraries/AP_Camera/AP_Camera.cpp
> <https://github.com/diydrones/ardupilot/pull/3194#discussion_r45167060>:
>
> >      if (get_distance(loc, _last_location) < _trigg_dist) {
> >          return false;
> >      }
> > -    _last_location = loc;
> > -    return true;
> > +
> > +    if ((unsigned) ahrs.roll_sensor/100.0f > _max_roll) {
>
> @magicrub <https://github.com/magicrub> it's a int32, so I don't think
> you want fabsf()....
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/diydrones/ardupilot/pull/3194/files#r45167060>.
>